lang/java/compiler/src/main/javacc/org/apache/avro/compiler/idl/idl.jj

```diff
@@ -150,12 +164,15 @@ public class Idl implements Closeable {
         throw error(key+" values must be textual: "+n, token);
     return values;
   }
-  
+
   private URL findFile(String importFile) throws IOException {
-    File file = new File(this.inputDir, importFile);
+    URI uri = this.inputDir.resolve(importFile);
+    File file = "file".equals(uri.getScheme()) ? new File(uri.getPath()) : null;
     URL result = null;
-    if (file.exists())
-      result = file.toURL();
+    if (file != null && file.exists())
+      result = file.toURI().toURL();
+    else if("classpath".equals(uri.getScheme()))
+      result = this.resourceLoader.getResource(uri.getPath().substring(1));
     else if (this.resourceLoader != null)
```

Comment:
The check here suggests that `this.resourceLoader` could be `null`. It seems now line 175 could lead to NullPointerException.